### PR TITLE
Cache `uidNumber` call

### LINKED ISSUE #1: Cache `uidNumber` call
It spawns a child process which is slow. The result should be cached for the lifetime of the process in case a user uses it multiple times.

https://github.com/sindresorhus/gulp-chown/blob/46a77e68572a54aa06d4d30c453b9c4b4ee920a7/index.js#L30

### PR DESCRIPTION
**Pull request by kevva**
Tried logging inside `uidNumber()` and it only ran once using `gulp.watch()` with multiple files and repeatedly saving them.

Fixes #1.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3,11 +3,13 @@ var gutil = require('gulp-util');
...
 var defaultMode = 511 & (~process.umask()); // 511 = 0777
+var cachedUid;
+var cachedGid;
 
 module.exports = function (user, group) {
 	var firstFile = true;
-	var finalUid = typeof user === 'number' ? user : null;
-	var finalGid = typeof group === 'number' ? group : null;
+	var finalUid = cachedUid || typeof user === 'number' ? user : null;
+	var finalGid = cachedGid || typeof group === 'number' ? group : null;
```
Comment (sindresorhus) [inline index.js]: This won't work if any are `0`.

Comment (kevva) [inline index.js]: Ah, true. Will fix. That's why tests were failing.

Comment (kevva) [inline index.js]: ~~A check of `< 0` should be sufficient, right?~~

Maybe not http://en.wikipedia.org/wiki/User_identifier#Special_values.

Comment (sindresorhus) [inline index.js]: `typeof cachedUid === 'number'`

Comment (kevva) [inline index.js]: Yeah, but so long lines :P. I don't want to wrap them in if statements. But whatever, long lines ftw.

**Comment by kevva** [comment]
lol now.

File: index.js [hunk 2]
```diff
@@ -18,21 +20,21 @@ module.exports = function (user, group) {
 		file.stat = file.stat || {};
 		file.stat.mode = file.stat.mode || defaultMode;
 
-		var finish = function () {
-			file.stat.uid = finalUid != null ? finalUid : file.stat.uid;
-			file.stat.gid = finalGid != null ? finalGid : file.stat.gid;
+		function finish() {
+			file.stat.uid = finalUid !== null ? finalUid : file.stat.uid;
+			file.stat.gid = finalGid !== null ? finalGid : file.stat.gid;
```
Comment (sindresorhus) [inline index.js]: Why?

Comment (kevva) [inline index.js]: The strict check or even checking them at all?

Comment (sindresorhus) [inline index.js]: Strict check

Comment (kevva) [inline index.js]: Removed.

Comment (sindresorhus) [inline index.js]: No, I was asking if it was changed for a reason because doing so changes the semantics of the check.

Comment (kevva) [inline index.js]: No reason, but might be a good idea to not be strict since we aren't setting it to null anywhere.

Comment (kevva) [inline index.js]: No reason, but since we are setting it to `null` on lines 11-12 it made sense.

**Comment by sindresorhus** [comment]
Awesome! Thanks again for fixing this :)

**Pull request merged**